lib/trino-parquet/src/main/java/io/trino/parquet/reader/UuidColumnReader.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.parquet.reader;
+
+import io.airlift.slice.Slice;
+import io.trino.parquet.RichColumnDescriptor;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.Type;
+import org.apache.parquet.io.api.Binary;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.airlift.slice.Slices.wrappedBuffer;
+import static io.trino.spi.type.UuidType.UUID;
+
+public class UuidColumnReader
+        extends PrimitiveColumnReader
+{
+    public UuidColumnReader(RichColumnDescriptor descriptor)
+    {
+        super(descriptor);
+    }
+
+    @Override
+    protected void readValue(BlockBuilder blockBuilder, Type trinoType)
+    {
+        checkArgument(trinoType == UUID, "Unsupported type: %s", trinoType);
```

Comment:
If we end-up here will the error message be understandable to user? Or should we assume it is not expected as would happen only in the case when parquet file has unannotated `FIXED_LEN_BYTE_ARRAY` column?

Btw I am missing context why `PrimitiveColumnReader` does not get as input what Trino type it is supposed to read. I expect this is not incidental. But it make it hard for me to assess if this PR is in-line with the design principles of Parquet reader.

Reply:
> Btw I am missing context why `PrimitiveColumnReader` does not get as input what Trino type it is supposed to read. I expect this is not incidental.

I don't think it's intentional. I think i was able to fix that in https://github.com/trinodb/trino/pull/2840 (see "Pass Presto Type to ColumnReader constructor"), but unfortunately i didn't finish that work.

> Or should we assume it is not expected as would happen only in the case when parquet file has unannotated FIXED_LEN_BYTE_ARRAY column?

Correct. Maybe i should renamed the reader class (without changing contents)?